tests/system_tests_grpc.py

```diff
@@ -0,0 +1,215 @@
+        "five@apache.org": ["two@apache.org", "four@apache.org"]
+    }
+
+    """
+    List of common friends to request for and the expected
+    result to be returned by the service
+    """
+    COMMON_FRIENDS_ITER = [
+        ["one@apache.org", "five@apache.org"],
+        ["two@apache.org", "four@apache.org"],
+    ]
+    COMMON_FRIENDS_EXP = [
+        ["two@apache.org", "four@apache.org"],
+        ["one@apache.org", "three@apache.org", "five@apache.org"],
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        super(GrpcTestBase, cls).setUpClass()
+        if skip_test():
+            return
+
+        # Define a random port for  the gRPC server to bind
+        cls.grpc_server_port = str(cls.tester.get_port())
+
+        # Run the gRPC server (see friendship.proto for more info)
+        cls.grpc_server = fs.serve(cls.grpc_server_port)
+
+        # Prepare router to communicate with the gRPC server
+        cls.connector_props = {
+            'port': cls.grpc_server_port,
+            'address': 'examples',
+            'host': '127.0.0.1',
+            'protocolVersion': 'HTTP2',
+            'name': 'grpc-server'
+        }
+        cls.router_http_port = cls.tester.get_port()
+        config = Qdrouterd.Config([
+            ('router', {'mode': 'standalone', 'id': 'QDR'}),
+            ('listener', {'port': cls.tester.get_port(), 'role': 'normal', 'host': '0.0.0.0'}),
+
+            ('httpListener', {'port': cls.router_http_port, 'address': 'examples',
+                              'host': '127.0.0.1', 'protocolVersion': 'HTTP2'}),
+            ('httpConnector', cls.connector_props)
+        ])
+        cls.router_qdr = cls.tester.qdrouterd("grpc-test-router", config, wait=True)
+
+        # If you wanna try it without the router, set the grpc_channel directly to the grpc_server_port
+        cls.grpc_channel = grpc.insecure_channel('127.0.0.1:%s' % cls.router_http_port)
+        cls.grpc_stub = FriendshipStub(cls.grpc_channel)
+
+    @classmethod
+    def tearDownClass(cls):
+        super(GrpcTestBase, cls).tearDownClass()
+        if skip_test():
+            return
+        cls.grpc_server.stop(TIMEOUT)
+
+    @classmethod
+    def create_person(cls, name, email):
+        p = Person()
+        p.name = name
+        p.email = email
+        res = cls.grpc_stub.Create(p)
+        assert res.success
+        assert res.message == ""
+        return p
+
+    @classmethod
+    def friendship_generator(cls):
+        for key in cls.FRIENDS:
+            for friend in cls.FRIENDS[key]:
+                fr = FriendshipRequest()
+                fr.email1 = key
+                fr.email2 = friend
+                yield fr
+
+    @classmethod
+    def common_friends_list(cls, friends):
+        for friend in friends:
+            pe = PersonEmail()
+            pe.email = friend
+            yield pe
+
+    @SkipIfNeeded(skip_test(), "grpcio is needed to run grpc tests")
```

Comment:
you should skip the entire class. https://docs.python.org/2.7/library/unittest.html#skipping-tests-and-expected-failures We can use all this now that we don't support Python 2.6 any more.

either put `@unittest.skipIf` on the class or throw the appropriate exception in `setUp()`.

Reply:
Maybe we can come up later (in an upcoming PR), with a definite solution to replace SkipIfNeeded,
since it was created exactly because of the lack of skipping tests with Python 2.6.

Reply:
Very well. If we leave the `SkipIfNeeded` in the test and stay compatible with 2.6, then we can maybe merge this right now.